internal/backend/grpc/validate.go

```diff
@@ -70,3 +83,88 @@ func validatePGPPublicKey(armored []byte, opts ...pgp.ValidationOption) (publicK
 		expiration: expiration,
 	}, nil
 }
+
+func (s *managementServer) ValidateJSONSchema(ctx context.Context, request *management.ValidateJsonSchemaRequest) (*management.ValidateJsonSchemaResponse, error) {
+	if _, err := auth.CheckGRPC(ctx, auth.WithValidSignature(true)); err != nil {
+		return nil, err
+	}
+
+	s.jsonSchemaCacheMu.Lock()
+	defer s.jsonSchemaCacheMu.Unlock()
+
+	shaSum := sha256.Sum256([]byte(request.Schema))
+	shaSumString := hex.EncodeToString(shaSum[:])
+
+	var err error
+
+	if s.jsonSchemaCache == nil {
+		s.jsonSchemaCache = map[string]*jsonschema.Schema{}
```

Comment:
I know that we require auth for this function, but a potential attack vector could be posting many different large schemas to this endpoint. Anything I post would be kept in memory forever. Maybe we should limit the cache size? (maybe use a library for it - like the cost-based eviction here or sth https://github.com/hypermodeinc/ristretto#features)

Reply:
used expirable.LRU without expiration but with the size limit.
We use it for the Talos clients already.